src/angular-app/languageforge/lexicon/settings/configuration/field-unified-view.model.ts

```diff
@@ -96,16 +95,47 @@ export class ConfigurationFieldUnifiedViewModel {
     }
   }
 
-  static selectAllRow(setting: SettingsBase, settings: SettingsBase[], selectAll: SettingsBase): void {
+  static ensureRequiredFields(fieldSettings: FieldSettings): boolean {
+    return RequiredFields.requiredFieldList().includes(fieldSettings.fieldName);
+  }
+
+  static selectAllRow(fieldSettings: FieldSettings, settings: SettingsBase[], selectAll: SettingsBase): void {
```

Comment:
You can't change the type from SettingsBase to FieldSettings here, because input systems settings also use this function. See my comment on the selectAllGroupColumn method for more details.

Reply:
The above was reverted back.I added the fieldName property to SettingsBase as both FieldSettings and InputSettings require it now.